packages/mobile/src/Components/Button.tsx

```diff
@@ -95,54 +95,93 @@ function withSizes(size: ButtonSize | undefined = 'medium', theme: any) {
 const Button: React.FunctionComponent<Omit<ButtonProps, 'height' | 'width'>> = (
   props: Omit<ButtonProps, 'height' | 'width'>
 ) => {
-  const { theme } = props;
+  const {
+    theme,
+    icon,
+    colorType,
+    mode,
+    elevation,
+    disabled,
+    size: propsSize,
+  } = props;
+
   const themeColors = {
     primary: theme.colors.primary,
     secondary: theme.colors.secondary,
     disabled: theme.colors.disabled,
     text: theme.colors.text,
     default: '#E0E0E0',
   };
-  let color = themeColors.primary;
-  let borderColor = theme.colors.primary;
-  let style = {};
 
-  if (props.colorType) {
-    color = themeColors[props.colorType];
-    borderColor = themeColors[props.colorType];
-  }
+  const color = React.useMemo(() => {
+    let colorReturn = themeColors.primary;
+
+    if (colorType) {
+      colorReturn = themeColors[colorType];
+    }
+
+    if (mode === 'outlined' || (mode === 'text' && colorType === 'default')) {
+      colorReturn = themeColors.text;
+    }
+
+    if (disabled) {
+      colorReturn = themeColors.disabled;
+    }
+
+    return colorReturn;
+  }, [colorType, mode, disabled]);
 
-  if (props.mode === 'outlined') {
-    color = themeColors.text;
-  }
+  const borderColor = React.useMemo(() => {
+    let borderColorReturn = theme.colors.primary;
 
-  if (props.mode === 'text' && props.colorType === 'default') {
-    color = themeColors.text;
-  }
+    if (colorType) {
+      borderColorReturn = themeColors[colorType];
+    }
 
-  if (props.elevation) {
-    style = elevationShadowStyle(props.elevation);
-  }
+    if (disabled) {
+      borderColorReturn = themeColors.disabled;
+    }
 
-  if (props.disabled) {
-    color = themeColors.disabled;
-    borderColor = themeColors.disabled;
-    style = {};
-  }
+    return borderColorReturn;
+  }, [colorType, disabled]);
+
+  const style = React.useMemo(
+    () => (elevation ? elevationShadowStyle(elevation) : {}),
+    [elevation]
+  );
+
+  const labelStyleMargin = React.useMemo(
+    () => ({
+      marginTop: 0,
+      marginRight: 0,
+      marginBottom: 0,
+      marginLeft: icon ? -4 : 0,
+    }),
+    [icon]
+  );
+
+  const contentStyleMargin = React.useMemo(
+    () => ({ marginLeft: icon && propsSize === 'small' ? -4 : 0 }),
```

Comment:
What is -4 here and on line 158? Declare this number with a meaningful name.

Reply:
Include margin types to better understanding